### PR TITLE
Api add migration or creating the tables for db

### PR DESCRIPTION
**Pull request by Elisei-26**

### CODE REVIEW

File: src/services/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@Injectable()
+export class UserService {
+	constructor(
+		@InjectRepository(User)
+		private readonly userRepository: Repository<User>,
+	) {}
+
+	async findAll(): Promise<User[]> {
+		const users: User[] = await this.userRepository.find();
+		return users;
```
Comment (danielursu) [inline src/services/user.service.ts]:
```suggestion
		return await this.userRepository.find();
```
let's simplify this

File: src/services/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	}
+
+	async getNumberOfUsers(): Promise<number> {
+		const [users, nr] = await this.userRepository.findAndCount();
+		return users.length;
+	}
```
Comment (danielursu) [inline src/services/user.service.ts]: this was for demo purposes. You can get rid of

File: src/app.controller.ts [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+
+	@Get("all")
+	getAll(): Promise<User[]> {
+		return this.userService.findAll();
+	}
+
+	@Get("number")
+	getNumberOfUsers(): Promise<number> {
+		return this.userService.getNumberOfUsers();
+	}
```
Comment (danielursu) [inline src/app.controller.ts]: you can delete this

**Review by danielursu — CHANGES_REQUESTED** [review]

File: src/app.controller.ts [hunk 4]
```diff
@@ -1,12 +1,19 @@
 import { Controller, Get } from "@nestjs/common";
 import { AppService } from "./app.service";
+import { User } from "./entities/user.entity";
+import { UserService } from "./services/user.service";
 
 @Controller()
```
Comment (danielursu) [inline src/app.controller.ts]: please add UserController
that will called when you enter /user

@Controller("user")

File: src/services/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+@Injectable()
+export class UserService {
+	constructor(
+		@InjectRepository(User)
+		private readonly userRepository: Repository<User>,
+	) {}
+
+	async findAll(): Promise<User[]> {
+		return await this.userRepository.find();
```
Comment (danielursu) [inline src/services/user.service.ts]: please add UserMapperService with toDTO() method
that method will transform your entity to a UserDTO interface

**Review by danielursu — CHANGES_REQUESTED** [review]

File: src/app.module.ts [hunk 6]
```diff
@@ -1,9 +1,28 @@
 import { Module } from "@nestjs/common";
 import { AppController } from "./app.controller";
 import { AppService } from "./app.service";
+import { TypeOrmModule } from "@nestjs/typeorm";
+import { ConfigModule } from "@nestjs/config";
+import { UserService } from "./user/user.service";
+import { User } from "./user/user.entity";
+import { UserMapper } from "./user/user.mapper";
+import { UserController } from "./user/user.controller";
```
Comment (danielursu) [inline src/app.module.ts]: delete unused imports

File: src/user/user.controller.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { Controller, Get } from "@nestjs/common";
+import { UserService } from "./user.service";
+import { User } from "./user.entity";
+
+@Controller("users")
```
Comment (danielursu) [inline src/user/user.controller.ts]:
```suggestion
@Controller("user")
```

File: src/user/user.controller.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+export class UserController {
+	constructor(private readonly userService: UserService) {}
+
+	@Get()
```
Comment (danielursu) [inline src/user/user.controller.ts]:
```suggestion
	@Get("all")
```

**Review by danielursu — CHANGES_REQUESTED** [review]

**Review by danielursu — APPROVED** [review]

**Review by samuelburaga — APPROVED** [review]

**Pull request merged**